api/src/utils/apply-query.ts

```diff
@@ -811,7 +838,12 @@ export function applyAggregate(dbQuery: Knex.QueryBuilder, aggregate: Aggregate,
 			}
 
 			if (operation === 'countDistinct') {
-				dbQuery.countDistinct(`${collection}.${field}`, { as: `countDistinct->${field}` });
+				if (!hasJoins && schema.collections[collection]?.primary === field) {
+					// Optimize to count as primary keys are unique
+					dbQuery.count(`${collection}.${field}`, { as: `countDistinct->${field}` });
+				} else {
+					dbQuery.countDistinct(`${collection}.${field}`, { as: `countDistinct->${field}` });
+				}
 			}
```

Comment:
This is the main logic change that optimises the `countDistinct` to a `count` while still returning the result as a `countDistinct`.

Used an inner conditional check to keep it simple and be consistent with the `count` operation:

https://github.com/directus/directus/blob/6092455e2630b2aa5b15f1b9fdcb4f13987790a6/api/src/utils/apply-query.ts#L832-L838